modules/extensions-spring-support/src/main/java/org/mule/runtime/module/extension/internal/config/dsl/ExtensionDefinitionParser.java

```diff
@@ -940,4 +935,25 @@ protected void parseInlineParameterGroup(ParameterGroupModel group)
   protected Optional<String> getInfrastructureParameterName(MetadataType fieldType) {
     return getId(fieldType).map(infrastructureParameterMap::get);
   }
+
+  private TypeConverter getTypeConverter(String name, MetadataType type, Object defaultValue, ExpressionSupport expressionSupport,
+                                         boolean required, Set<ModelProperty> modelProperties, boolean acceptsReferences,
+                                         boolean content) {
+
+    ClassLoader extensionClassLoader = getContextClassLoader();
+    TypeConverter typeConverter = value -> {
+      Thread thread = Thread.currentThread();
+      ClassLoader currentClassLoader = thread.getContextClassLoader();
+      CompositeClassLoader compositeClassLoader = from(extensionClassLoader, currentClassLoader);
+      setContextClassLoader(thread, currentClassLoader, compositeClassLoader);
+      try {
+        return valueResolverFactory.of(name, type, value, defaultValue, expressionSupport, required, modelProperties,
+                                       acceptsReferences, content);
+      } finally {
+        setContextClassLoader(thread, extensionClassLoader, currentClassLoader);
```

Comment:
one question, shouldn't it be` setContextClassLoader(thread, compositeClassLoader, currentClassLoader);`